### PR TITLE
Feature/gdpo

### LINKED ISSUE #977: Add GDPO Support
**Is your feature request related to a problem? Please describe.**

 GRPO inherently compresses the reward signal, causing loss of information in the advantage estimates. Group-wise decoupled normalization of each reward separately better preserves cross-reward distinctions and enables more accurate multi-reward optimization.

**Describe the solution you'd like**
https://arxiv.org/abs/2601.05242

**Additional context**
Authors have reported huge improvement in results

How i came to this  --? 
As part of Google Tunix hackathon i have observed the problem of GRPO with a multi reward optimization. I found this latest paper directly questions the issues of loss of information in advantage estimates in GRPO hence i would love to add this to Tunix!

**Checklist**

- [x] I have searched the existing issues for similar feature requests.
- [x] This is not a support question (please use the "bug template" for that).

### PR DESCRIPTION
**Pull request by kmsgnnew**
Resolves #977 

https://github.com/google/tunix/issues/977

Added implementation of GDPO. Shared below reference link and implementation. This paper improves upon GRPO.

**Reference**
Reference paper
https://arxiv.org/abs/2601.05242
Reference Implementation 
https://github.com/NVlabs/GDPO/blob/b080e63d0126870ad08acc8ebc3f04b728175a9e/trl-GDPO/trl-0.18.0-gdpo/trl/trainer/grpo_trainer.py#L1222

**Colab Notebook**
NiL

**Checklist**

- [x] I have added all the necessary unit tests for my change.
- [x] I have verified that my change does not break existing code and all unit tests pass.
- [x] I have added all appropriate doc-strings/documentation.
- [x] My PR is based on the latest changes of the main branch (if unsure, rebase the code).
- [x] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).
- [x] I have followed [Contribution Guidelines](https://github.com/google/tunix/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: tunix/rl/grpo/gdpo_learner.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    each_reward_mean_grouped = each_reward_mean_grouped.repeat(num_generations)
+    each_reward_std_grouped = each_reward_std_grouped.repeat(num_generations)
+    each_reward_advantage = reward_for_index - each_reward_mean_grouped
+    each_reward_advantage = each_reward_advantage / (
+        each_reward_std_grouped + 1e-4
+    )
+    all_reward_advantage.append(each_reward_advantage)
+
+  combined_reward_advantage = jnp.stack(all_reward_advantage, axis=1)
+  pre_bn_advantages = jnp.nansum(combined_reward_advantage, axis=1)
```
[the remark this answers is not in the document]
Comment (kmsgnnew) [reply tunix/rl/grpo/gdpo_learner.py]: updated

File: tunix/rl/grpo/gdpo_learner.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+      # Log all metrics for this trajectory in one call
+      if step is not None:
+        self.rl_cluster.buffer_metrics_async(
+            metrics_to_log, mode=mode, step=step
+        )
+      else:
+        self.rl_cluster.buffer_metrics(metrics_to_log, mode=mode)
+
+    return jnp.array(rewards)
```
[the remark this answers is not in the document]
Comment (kmsgnnew) [reply tunix/rl/grpo/gdpo_learner.py]: implemented _generate_and_compute_advantage similar from grpo but used a new function _compute_gdpo_specific_rewards
